### DISCUSSION TITLE
Guide to documentation and some questions related to it

### DISCUSSION THREAD

**Discussion by jj10133** [osmandapp/OsmAnd-iOS#3609] [General]
Hi team super new to learning swift and swiftui however I love writing the UI in swift i don't know why.. was primarily as backend guy. Anyway I will be honest I love OSMAnd however the UI is not great, plus following Swift 6.0  I want to change the UI for the app and same time learn more about SwiftUI. I don't know anything under the hood how maps are working, could someone please guide to documentation or give a brief overview on this?

**Comment by vshcherb** [comment]
That might be quite difficult cause we didn't migrate from Objective-C to Swift + old libraries + tons of code in C++. It's certainly a hard project to start with

**Reply by jj10133** [reply]
Is OsmAnd Core the main base of it?

**Reply by vshcherb** [reply]
Yes

**Reply by jj10133** [reply]
Tbh never heard of  objective c++... im super new to swift even but im trying understand the objective c code on how its calling the c++ code. I will give try on setting up the c++ library in swift. I have worked with creating a wrapper around c lib in swift but recently i got to know interoperability with c++ so im like can try to use swiftui

**Reply by jj10133** [reply]
At the moment goal is to show maps but super lost... 🤣🤣🤣

**Comment by jj10133** [comment]
@vshcherb I'm back again here!! this time with lot of experiments and making some p2p apps ofcourse with swift and have even made p2p maps that serves tiles using hypercore "just blocks of tile" -> peers download only tiles based on xyz and serve them nearby!!


I'm hoping to touch again everything and try to code since I know that we can use C++ code in swift however idk how much interopable is it
